### ISSUE TITLE
[geonetwork] es/kibana settings not taken from geonetwork.properties

### ISSUE THREAD

**Issue by landryb** [georchestra/georchestra#3655]
unsure for `kb.url` but testing an alternative `es.port` in `/etc/georchestra/geonetwork/geonetwork.properties` shows that GN4 keeps trying to contact port 9200. Turns out `es.port` (and other es/kibana-related config keys) is also defined in `WEB-INF/config.properties` and it seems this value takes priority over the one defined in the georchestra datadir.

**Comment by landryb** [comment]
my understanding is that the file shipped in the webapp comes from https://github.com/georchestra/geonetwork/blob/georchestra-gn4-4.0.6/web/src/main/webResources/WEB-INF/config.properties and the variable values seems to come from https://github.com/georchestra/geonetwork/blob/georchestra-gn4-4.0.6/pom.xml

ideally all the es/kibana variables should be taken from `geonetwork.properties` to allow proper overrides from an admin, for example if one wants to set a login/password between gn4 and es (which is recommended by the es production settings docs) or say store in a single es cluster indexes coming from distinct gn4 instances (dev, staging, prod...) and name the indexes differently from `gn-records`, default value for `es.index.records`

**Comment by landryb** [comment]
something missing/wrong in f2b002320e ? or my testing was wrong... @pmauduit ?

**Comment by landryb** [comment]
it seems the [WEB-INF/config.properties](https://github.com/georchestra/geonetwork/blob/georchestra-gn4-4.0.6/web/src/main/webResources/WEB-INF/config.properties) file shipped in the webapp has values from web.xml instead of the `${}` placeholders, so something in the build system replaces the placeholders ?

**Comment by landryb** [comment]
i've tried setting `es.port=${es.port}` in `WEB-INF/config.properties` but that badly blows at startup:
```
<2>Caused by: java.lang.IllegalArgumentException: Circular placeholder reference 'es.port' in property definitions
<2>    at org.springframework.util.PropertyPlaceholderHelper.parseStringValue(PropertyPlaceholderHelper.java:145)
<2>    at org.springframework.util.PropertyPlaceholderHelper.parseStringValue(PropertyPlaceholderHelper.java:166)
<2>    at org.springframework.util.PropertyPlaceholderHelper.parseStringValue(PropertyPlaceholderHelper.java:166)
```

another option that i tried was just to comment out the values in `WEB-INF/config.properties` (eg say `es.port` & `es.url` which references `es.port`) - with this, the values coming from `/etc/georchestra/geonetwork/geonetwork.properties` should be used by GN4, but that doesnt work, as i guess some parts of GN4 don't know to take values from `geonetwork.properties` and blow when doing a search:
```
java.net.MalformedURLException: no protocol: ${es.url}/gn-records-demo/_search?
        at java.net.URL.<init>(URL.java:611)
        at java.net.URL.<init>(URL.java:508)
        at java.net.URL.<init>(URL.java:457)
        at org.fao.geonet.api.es.EsHTTPProxy.handleRequest(EsHTTPProxy.java:470)
        at org.fao.geonet.api.es.EsHTTPProxy.call(EsHTTPProxy.java:334)
        at org.fao.geonet.api.es.EsHTTPProxy.search(EsHTTPProxy.java:258)
        at org.fao.geonet.api.es.EsHTTPProxy$$FastClassBySpringCGLIB$$bdd2f76a.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:687)
```

**Comment by landryb** [comment]
@pmauduit to summarize, i think to solve this issue we:
- should remove the values overriden by `geonetwork.properties` from `config.properties`
- move `es.url` definition from `config.properties` to `geonetwork.properties` as it references `es.port`

that doesn't explain with properties priority ordering doesnt seem to work..

**Comment by pmauduit** [comment]
Looking at config.properties coming from the classpath, it is loaded from the following xml files:

* `config-spring-geonetwork.xml:41:  <context:property-placeholder location="WEB-INF/config.properties"`
* `spring-servlet.xml:45:  <context:property-placeholder location="WEB-INF/config.properties"`

If we want the properties file from the datadir being able to override this one, then we should make sure it is also loaded into the same files. For both files, it seems to be the case already.

**Comment by pmauduit** [comment]
> that doesn't explain with properties priority ordering doesnt seem to work..

I think I read that this evolved across the spring versions, and we already had to dig some issues ressembling to this one. It's not completely clear to me either.

**Comment by pmauduit** [comment]
I'm wondering if the "wrongly" substituted variables are not coming from here:
https://github.com/georchestra/geonetwork/blob/georchestra-gn4-4.0.6/index/src/main/java/org/fao/geonet/index/es/EsRestClient.java#L89-L105

The bean is declared in the following xml:
https://github.com/georchestra/geonetwork/blob/georchestra-gn4-4.0.6/core/src/main/resources/config-spring-geonetwork.xml#L92-L93

where the properties file from the datadir is not sourced:
https://github.com/georchestra/geonetwork/blob/georchestra-gn4-4.0.6/core/src/main/resources/config-spring-geonetwork.xml#L36-L38

Fortunately, we already override the resulting jar (suffixed the version with "-georchestra") for the core module:
https://github.com/georchestra/geonetwork/blob/georchestra-gn4-4.0.6/core/pom.xml#L30

So I think that adding the properties file from the dadadir in this module should solve our issue.

**Comment by landryb** [comment]
fixed by georchestra/geonetwork#202
